arm-dns/2016-04-01/swagger/dns.json

```diff
@@ -0,0 +1,1202 @@
+          "description": "Gets or sets the type of the resource."
+        },
+        "etag": {
+          "type": "string",
+          "description": "Gets or sets the ETag of the RecordSet."
+        },
+        "location": {
+          "type": "string",
+          "description": "Gets or sets the location of the resource."
+        },
+        "properties": {
+          "$ref": "#/definitions/RecordSetProperties",
+          "description": "Gets or sets the properties of the RecordSet."
+        }
+      },
+      "description": "Describes a DNS RecordSet (a set of DNS records with the same name and type)."
+    },
+    "RecordSetUpdateParameters": {
+      "properties": {
+        "RecordSet": {
+          "$ref": "#/definitions/RecordSet",
+          "description": "Gets or sets information about the RecordSet being updated."
+        }
+      },
+      "description": "Parameters supplied to update a RecordSet."
+    },
+    "RecordSetCreateOrUpdateParameters": {
+      "properties": {
+        "RecordSet": {
+          "$ref": "#/definitions/RecordSet",
+          "description": "Gets or sets information about the RecordSet being created or updated."
+        }
+      },
+      "description": "Parameters supplied to create or update a RecordSet."
+    },
+    "RecordSetListResult": {
+      "properties": {
+        "value": {
+          "type": "array",
+          "items": {
+            "$ref": "#/definitions/RecordSet"
+          },
+          "description": "Gets or sets information about the RecordSets in the response."
+        },
+        "nextLink": {
+          "type": "string",
+          "description": "Gets or sets the continuation token for the next page."
+        }
+      },
+      "description": "The response to a RecordSet List operation."
+    },
+    "ZoneProperties": {
+      "properties": {
+        "maxNumberOfRecordSets": {
+          "type": "integer",
+          "format": "int64",
+          "description": "Gets or sets the maximum number of record sets that can be created in this zone."
+        },
+        "numberOfRecordSets": {
+          "type": "integer",
+          "format": "int64",
+          "description": "Gets or sets the current number of record sets in this zone."
+        },
+        "nameServers": {
+          "type": "array",
+          "items": {
+            "type": "string"
+          },
+          "description": "Gets the name servers populated for this zone. This is a read-only property and any attempt to set this value will be ignored."
+        }
+      },
+      "description": "Represents the properties of the zone."
+    },
+    "Zone": {
+      "properties": {
+        "etag": {
+          "type": "string",
+          "description": "Gets or sets the ETag of the zone that is being updated, as received from a Get operation."
+        },
+        "properties": {
+          "$ref": "#/definitions/ZoneProperties",
+          "description": "Gets or sets the properties of the zone."
```

Comment:
That looks like a bug on the RP side. If none of the ZoneProperties are provided why would the server expect to provide an empty `{ properties: {} }` object?